### DISCUSSION TITLE
Dozens of duplicated events on the html element

### DISCUSSION THREAD

**Discussion by o-alquimista** [twbs/bootstrap#33980] [Q&A]
Is this normal?

![Captura de tela de 2021-05-14 00-04-56](https://user-images.githubusercontent.com/25424263/118215541-0df90280-b448-11eb-8b22-bca2dc4e48c8.png)

Bootstrap 5.0.0

[the remark this answers is not in the document]
**Reply by rohit2sharma95** [reply]
Click events are necessary for some plugins, but at least this DOMContentLoaded event can be reduced somehow IMO.

[the remark this answers is not in the document]
**Reply by alpadev** [reply]
Yeah that's normal. Every component that can be used through jQuery registers itself individually within a `DOMContentLoaded` event. It reuses the same lines of code to achieve this. That's why all those events point to the same place in the file. So they are not duplicates per se, but rather multiple invocations.

Not that it's really a problem but agreeing with @rohit2sharma95, we could refactor this to attach just one `DOMContentLoaded` event and register all components at once.

[the remark this answers is not in the document]
**Reply by rohit2sharma95** [reply]
Created a new issue for reference #34103.
